e2e/test/scenarios/models/reproductions.cy.spec.ts

```diff
@@ -837,6 +841,47 @@ describe("issue 39993", () => {
   });
 });
 
+describe("issue 33844", () => {
```

Comment:
This symptom is also present in `master`.
LMK if you think we should fix this separately.

----

1. New -> Model -> Orders
2. Metadata -> Select ID -> Detail views only -> Save
3. Make sure the ID column is hidden in Simple mode now
4. Open Metadata Editor again. ID column should be visible and the value should "Detail views only".
5. Cancel editing, discard changes

ID column is visible in simple mode

6. Open Metadata Editor again.

ID column has "Table and details views" setting.

----

https://github.com/user-attachments/assets/98dce98d-7380-459a-bd59-c156bd30dcf1